src/lib/OAuthProvider.ts

```diff
@@ -58,10 +58,12 @@ export class OAuthProvider {
 
 		if (this.useFileCache) {
 			try {
-				fs.accessSync(OAuthProvider.cacheDir, fs.constants.W_OK)
+				if (!fs.existsSync(OAuthProvider.cacheDir)) {
```

Comment:
If it does exist we should check that it is writable.

Reply:
Added check